electroncash_gui/qt/installwizard.py

```diff
@@ -603,3 +634,106 @@ def do_dialog():
                                           "After selecting a server, select a wallet file to open."))
             QTimer.singleShot(10, do_dialog)
         return ret
+
+class DerivationPathScanner(QThread):
+
+    DERIVATION_PATHS = [
+        "m/44'/145'/0'",
+        "m/144'/44'/0'",
+        "m/144'/0'/0'",
+        "m/44'/0'/0'/0",
+        "m/0'/0",
+        "m/0",
+        "m/0'/0'",
+        "m/44'/145'/0'/0",
+        "m/44'/245'/0"]
+
+    def __init__(self, parent, seed, progress_cb):
+        QThread.__init__(self, parent)
+        self.progress_cb = progress_cb
+        self.parent = parent
+        self.seed = seed
+        self.aborting = False
+
+    def run(self):
+        for i, p in enumerate(self.DERIVATION_PATHS):
+            if self.aborting:
+                return
+            from electroncash import keystore
+            from electroncash.wallet import Standard_Wallet
+            from electroncash.storage import WalletStorage
+            from electroncash.network import Network
+            k = keystore.from_seed(self.seed, '', derivation=p, seed_type=self.parent.seed_type)
+            storage_path = self.parent.config.get_wallet_path() + "_not_saved_on_disk"
+            tmp_storage = WalletStorage(storage_path, in_memory_only=True)
+            tmp_storage.put('seed_type', self.parent.seed_type)
+            keys = k.dump()
+            tmp_storage.put('keystore', keys)
+            wallet = Standard_Wallet(tmp_storage)
+            network = Network.get_instance()
+            if network:
+                wallet.start_threads(network)
+                wallet.synchronize()
+                wallet.wait_until_synchronized()
+                while network.is_connecting():
+                    time.sleep(0.1)
+                num_tx = len(wallet.get_history())
+                wallet.clear_history()
+                self.progress_cb(i, str(num_tx))
```

Comment:
Hey man; this is touching the GUI from a thread that is not the main thread.  That's UB. 

The typical way to do this is you leverage Qt's signal/slot mechanism to emit a signal which gets delivered to the main thread.  An example of this can be found in `main_window.py` where you have `def on_network():` which is executed as a callback in the network thread, and all that does is immediately emit a signal that gets caught in the main thread in `def on_network_qt():`...

Here is more background on how to work with Qt across threads:  https://doc.qt.io/qt-5/threads-qobject.html

> Although QObject is reentrant, the GUI classes, notably QWidget and all its subclasses, are not reentrant. They can only be used from the main thread. As noted earlier, QCoreApplication::exec() must also be called from that thread.
>
>In practice, the impossibility of using GUI classes in other threads than the main thread can easily be worked around ...

Reply:
> Hey man; this is touching the GUI from a thread that is not the main thread. That's UB.
Oh, I see. That shouldn't be too hard to work around and I'll see if I get time for it this weekend.
Thanks for the review.

Reply:
Alternatively I could also work on this and "finish it up" for you to handle this.  I like this feature..  and I think it's a valuable addition.  So. whomever makes it first wins I guess :)